### ISSUE TITLE
Addon structures will save with null data and brick the vehicle

### ISSUE THREAD

**Issue by SprocketTools** [Muushy/Sprocket-Feedback#638]
**Describe the bug**
Addon structures will often times save without any data, like this:
```
{
      "vuid": 4,
      "type": "plateStructureMesh",
      "meshData": {
        "v": "0.1",
        "smoothAngle": 0,
        "gridSize": 1,
        "format": "freeform",
        "mesh": {
          "majorVersion": 0,
          "minorVersion": 3,
          "vertices": [],
          "edges": [],
          "edgeFlags": [],
          "faces": []
        },
        "rivets": {
          "profiles": [
            {
              "model": 0,
              "spacing": 0.1,
              "diameter": 0.05,
              "height": 0.025,
              "padding": 0.04
            }
          ],
          "nodes": []
        }
      }
    },
```

**Version**
0.2.19.1

**To Reproduce**
The easiest way to reproduce this bug is to create a new design, place a standard addon structure down, then save.  The blueprint will have a null structure, while reloading the file deletes the structure.  I've also observed these in other vehicles that are unable to load.

**Comment by SprocketTools** [comment]
![image](https://github.com/user-attachments/assets/df290b9b-e1a0-49af-9b90-7bdea92e6c18)
Actually, it appears that duplicates were saved here

**Comment by SprocketTools** [comment]
Similar behavior has also been found with addon structure "blueprints" and "objects" - phantom items will get saved into the blueprint despite not existing in the tank.

**Comment by Muushy** [comment]
Fixed in 0.2.19.3
